torch/nn/parameter.py

```diff
@@ -44,3 +44,93 @@ def __reduce_ex__(self, proto):
         )
 
     __torch_function__ = _disabled_torch_function_impl
+
+
+class UninitializedParameter(Parameter):
+    r"""A parameter that can be lazily initialized.
+
+    Uninitialized Parameters holds empty tensors that can be moved
```

Comment:
Start by briefly describing how UninitializedParameters are like Parameters, then describe how they are different: basically that many operations supported on normal parameters are not supported on UninitializedParameters.

Reply:
Updated ccomment